onnxruntime/test/providers/cpu/tensor/split_op_test.cc

```diff
@@ -56,6 +56,28 @@ TEST(SplitOperatorTest, Axis0EqualSplit) {
   RunTest(axis, {}, input, outputs);
 }
 
+TEST(SplitOperatorTest, Axis0EqualSplitInt32) {
+  const int64_t axis = 0;
+  std::vector<ShapeAndData> outputs;
+
+  // input shape and data
+  ShapeAndData input = {{4, 2},  // shape
```

Comment:
Isn't ShapeAndData a pair containing a vector<int64_t> and a vector<float> ? I think this will just cast the input data to floats and will not test your code path. You can consider making ShapeAndData templatized and instantiate it as 'float' everywhere else and 'int32_t' in your use-case.

Reply:
Thanks, just changed that :)